dll/win32/browseui/addresseditbox.cpp

```diff
@@ -130,17 +130,22 @@ HRESULT CAddressEditBox::RefreshAddress()
 
     /* Set the path if filesystem; otherwise use the name */
     WCHAR szPathOrName[MAX_PATH];
-    if (!SHGetPathFromIDListW(absolutePIDL, szPathOrName))
-    {
-        STRRET ret;
-        hr = pShellFolder->GetDisplayNameOf(pidlChild, SHGDN_FORADDRESSBAR | SHGDN_FORPARSING, &ret);
-        if (FAILED_UNEXPECTEDLY(hr))
-            return hr;
+    SHGDNF flags;
+
+    if (gCabinetState.fFullPathAddress)
+        flags = SHGDN_FORADDRESSBAR | SHGDN_FORPARSING;
+    else
+        flags = SHGDN_FORADDRESSBAR;
+
+    STRRET ret;
+    hr = pShellFolder->GetDisplayNameOf(pidlChild, flags, &ret);
```

Comment:
This kinds of look similar to IEGetNameAndFlags but also a bit different. Do you know whether using that other function would make the code behave differently?

Reply:
`IEGetNameAndFlags` looks like a wrapper for `GetDisplayNameOf`. I can expose the `IEGetNameAndFlags` function through a header and use it here instead to avoid code duplication.